server/src/main/java/org/opensearch/search/labels/rules/DefaultUserInfoLabelingRule.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.labels.rules;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.common.Strings;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Rules to get user info labels, specifically, the info is injected by the security plugin.
+ */
+public class DefaultUserInfoLabelingRule implements Rule{
+    /**
+     * Constant setting for user info header key that are injected during authentication
+     */
+    private final String REQUEST_HEADER_USER_INFO = "_opendistro_security_user_info";
+    /**
+     * Constant setting for remote address info header key that are injected during authentication
+     */
+    private final String REQUEST_HEADER_REMOTE_ADDRESS = "_opendistro_security_remote_address";
+
+    private final String REMOTE_ADDRESS = "remote_address";
+    private final String USER_NAME = "user_name";
+    private final String USER_BACKEND_ROLES = "user_backend_roles";
+    private final String USER_ROLES = "user_roles";
+    private final String USER_TENANT = "user_tenant";
+
+    /**
+     * @param threadContext
+     * @param searchRequest
+     * @return Map of User related info and the corresponding values
+     */
+    @Override
+    public Map<String, Object> evaluate(ThreadContext threadContext, SearchRequest searchRequest) {
+        // for POC testing only. This is injected by the security plugin
+        threadContext.putTransient(REQUEST_HEADER_USER_INFO, "chenyang|br1,b2|r1,r2|t1");
+        threadContext.putTransient(REQUEST_HEADER_REMOTE_ADDRESS, "1.2.3.4");
+        return getUserInfoFromThreadContext(threadContext);
+    }
+
+    /**
+     * Get User info, specifically injected by the security plugin, from the thread context
+     *
+     * @param threadContext context of the thread
+     * @return Map of User related info and the corresponding values
+     */
+    private Map<String, Object> getUserInfoFromThreadContext(ThreadContext threadContext) {
+        Map<String, Object> userInfoMap = new HashMap<>();
+        if (threadContext == null) {
+            return userInfoMap;
+        }
+        Object userInfoObj = threadContext.getTransient(REQUEST_HEADER_USER_INFO);
+        if (userInfoObj == null) {
+            return userInfoMap;
+        }
+        String userInfoStr = userInfoObj.toString();
+        Object remoteAddressObj = threadContext.getTransient(REQUEST_HEADER_REMOTE_ADDRESS);
+        if (remoteAddressObj != null) {
+            userInfoMap.put(REMOTE_ADDRESS, remoteAddressObj.toString());
+        }
+
+        String[] userInfo = userInfoStr.split("\\|");
```

Comment:
We can directly get the `User` object from commons package using this method: https://github.com/opensearch-project/common-utils/blob/main/src/main/java/org/opensearch/commons/authuser/User.java#L164

Reply:
make sense!

Reply:
That would bring the dependency of common-utils in core.

Reply:
yes, in this case we need to add common-utils as a dependency of core, which won't be possible. So I just copied the logic from common-utils here :)